docs: add an example how to replace the DefaultHttpUrlGenerator (#2278)
ngrx/platform: projects/ngrx.io/content/guide/data/extension-points.md

### Replace the HttpUrlGenerator

This example replaces the `DefaultHttpUrlGenerator` with a customized `HttpUrlGenerator` that pluralizes both collection resource and entity resource URLs.

The implementation simply overrides `DefaultHttpUrlGenerator.getResourceUrls(string, string)`:

```ts
import {
  DefaultHttpUrlGenerator,
  HttpResourceUrls,
  normalizeRoot,
  Pluralizer
} from '@ngrx/data';

export class PluralHttpUrlGenerator extends DefaultHttpUrlGenerator {
  constructor(private pluralizzer: Pluralizer) {
    super(pluralizzer);
  }

  protected getResourceUrls(
    entityName: string,
    root: string
  ): HttpResourceUrls {
    let resourceUrls = this.knownHttpResourceUrls[entityName];
    if (!resourceUrls) {
      const nRoot = normalizeRoot(root);
      const url = `${nRoot}/${this.pluralizzer.pluralize(
        entityName
      )}/`.toLowerCase();
      resourceUrls = {
        entityResourceUrl: url,
        collectionResourceUrl: url
      };
      this.registerHttpResourceUrls({ [entityName]: resourceUrls });
    }
    return resourceUrls;
  }
}
```

> In order to hook it into ngrx data we need to provide it in the module where ngrx data is imported:

Reviewer: ```suggestion
Override the `HttpUrlGenerator` provider in the root `AppModule` where `EntityDataModule.forRoot()` is imported:
```

Author: LGTM, but I'm not sure if I need to update the branch. Or will they come through automatically, when the change request is resolved?

Reviewer: If you accept the suggestions, it will update the branch

Reviewer: If you accept the suggestions, it will update the branch

```ts
@NgModule({
  // ...
  imports: [
    // ...
    EntityDataModule.forRoot({})
  ],
  providers: [
    // ...
    { provide: HttpUrlGenerator, useClass: PluralHttpUrlGenerator }
  ]
})
export class AppModule {}
```

> Don't forget the unit tests:

Reviewer: ```suggestion
To unit test the custom HTTP URL generator:
```
